### PR TITLE
Add a verification script

### PR DESCRIPTION
**Pull request by maelk**
verifications:
- Running kubernetes cluster, reachable
- Running baremetal operator
- Running CAPI provider baremetal operator
- CRDs defined
- Host vms defined
- Networking ok
- Ironic containers running
- baremetal hosts CRs created

### CODE REVIEW

File: 04_verify.sh [hunk 1]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+
+set -u
+
+exec 3>&1
+
+source lib/logging.sh
+source lib/common.sh
+
+KUBECONFIG=~/.kube/config
```
Comment (kanwar-saad) [inline 04_verify.sh]: this is already default path for KUBECONFIG. It could be
KUBECONFIG="${KUBECONFIG:-~/.kube/config}

File: 04_verify.sh [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+EXPTD_CRDS="baremetalhosts.metal3.io clusters.cluster.k8s.io \
```
Comment (kanwar-saad) [inline 04_verify.sh]: maybe nicer to put one crd in one line

File: 04_verify.sh [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+echo "" >&3
+
+# Verify the baremetal hosts
+## Fetch the BM CRs
+BM_HOSTS="$(kubectl --kubeconfig ${KUBECONFIG} get baremetalhosts -n metal3 -o json)"
+FAILS=$(process_status $? "Fetch Baremetalhosts")
+## Fetch the VMs
+BM_VMS="$(virsh list --all)"
+FAILS=$(process_status $? "Fetch Baremetalhosts VMs")
+## Verify
```
Comment (kanwar-saad) [inline 04_verify.sh:174]: one empty line before each step.

File: 04_verify.sh [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+#Verify that a resource exists in a type
+function check_k8s_entity() {
+  for name in ${2}; do
+    # Check entity exists
+    ENTITY="$(kubectl --kubeconfig "${KUBECONFIG}" get "${1}" "${name}" -n metal3 -o json)"
+    FAILS="$(process_status $? "${1} ${name} created")"
+    # Check the replicas
+    FAILS="$(equals "$(echo "${ENTITY}" | jq -r '.status.readyReplicas')" \
```
Comment (kanwar-saad) [inline 04_verify.sh:106]: FAILS is overwritten here without any check .. is that expected?

File: 04_verify.sh [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  done
+  echo "" >&3
+  echo "${FAILS}"
+}
+
+# Verify networking
+for bridge in ${BRIDGES}; do
+  ip link show dev "${bridge}" > /dev/null
+  FAILS=$(process_status $? "Network ${bridge} exists")
+  #Bridge baremetal is down, disabling checks on state
```
Comment (kanwar-saad) [inline 04_verify.sh]: remove commented code

File: 04_verify.sh [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+  done
+  echo "" >&3
+  echo "${FAILS}"
+}
+
+#Verify that a resource exists in a type
+function check_k8s_rs() {
+  for name in ${1}; do
+    # Check entity exists
+    ENTITY="$(kubectl --kubeconfig "${KUBECONFIG}" get replicasets -l name="${name}" -n metal3 -o json | jq '.items[0]')"
```
Comment (kanwar-saad) [inline 04_verify.sh:119]: you can put this line
kubectl --kubeconfig "${KUBECONFIG}"
to KUBECTL
to reduce clutter.

File: 04_verify.sh [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    # Check the replicas
+    FAILS="$(equals "$(echo "${ENTITY}" | jq -r '.status.readyReplicas')" \
+      "$(echo "${ENTITY}" | jq -r '.status.replicas')" \
+      "${name} ${1} replicas correct")"
+  done
+  echo "" >&3
+  echo "${FAILS}"
+}
+
+#Verify that a resource exists in a type
```
Comment (kanwar-saad) [inline 04_verify.sh:115]: same comment?

File: 04_verify.sh [hunk 8]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+
+set -u
+
+exec 3>&1
```
Comment (kanwar-saad) [inline 04_verify.sh:7]: maybe a comment would be useful here.

File: 04_verify.sh [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+source lib/logging.sh
+source lib/common.sh
+
+KUBECONFIG=~/.kube/config
+EXPTD_CRDS="baremetalhosts.metal3.io clusters.cluster.k8s.io \
+  machineclasses.cluster.k8s.io machinedeployments.cluster.k8s.io \
+  machines.cluster.k8s.io machinesets.cluster.k8s.io"
+EXPTD_STATEFULSETS="cluster-api-controller-manager \
```
Comment (kanwar-saad) [inline 04_verify.sh]: These definitions can be moved after function definitions so that all global stuff remains close.

File: 04_verify.sh (hunk 4 above)
Comment (maelk) [reply 04_verify.sh:106]: The equals function is a check in itself

**Pull request merged**